src/github_repo_request_local.py

```diff
@@ -135,6 +156,45 @@ def list_test_files(directory, excluded_extensions):
     return test_files
 
 
+def detect_test_runners(repo_path):
+    runner_presence = {runner: 0 for runner in test_runners}  # Initialise
+    # detection dictionary
+
+    # Search through files and directories
+    for root, dirs, files in os.walk(repo_path):
+        for runner, indicators in test_runners.items():
+            # Check for configuration files and patterns
+            for config_file in indicators["config_files"]:
+                if config_file in files:
+                    runner_presence[runner] += 1  # Increment if config file is
+                    # found
+
+            # Check for file patterns
+            for pattern in indicators["file_patterns"]:
+                matching_files = [file for file in files if re.match(pattern, file)]
+                runner_presence[runner] += len(
+                    matching_files
+                )  # Increment by number of matching test
+                # files
+
+            # Check for dependencies (if applicable)
+            for dep_pattern in indicators["dependency_patterns"]:
+                for file in files:
+                    path = os.path.join(root, file)
+                    if (
+                        os.path.exists(path) and file in indicators["config_files"]
+                    ):  # Ensure dependency checks in config
+                        # files
+                        with open(path, "r") as file_content:
```

Comment:
- introduced a function `def measure_performance(func, *args, **kwargs):` to capture `execution_time`, `elapsed_time`,`memory_used`, `cpu_usage_user`, `cpu_usage_system`, `cpu_usage_idle` but was getting some negative values.
- imported `import gc` garbage collection
- still getting negative values in some columns
- Decided to put this function in the `utility` folder so that it can be used later on for any function
- saving the performance df s to then compare them
- The running of the script is taking much longer